analyzers/src/SonarAnalyzer.Common/Helpers/MemberDescriptor.cs

```diff
@@ -50,8 +51,9 @@ public bool IsMatch<TSymbolType>(string memberName, Lazy<TSymbolType> memberSymb
         public static bool MatchesAny<TSymbolType>(string memberName, Lazy<TSymbolType> memberSymbol, bool checkOverriddenMethods, StringComparison nameComparison, params MemberDescriptor[] members)
             where TSymbolType : class, ISymbol =>
             memberName != null
+            && members.Any(x => memberName.Equals(x.Name, nameComparison))
```

Comment:
HasSameName does null-check inside, already done here

Reply:
Doesn't this change the functionality?

Previously we checked that the member with the same name had the correct containing type now we check if we have a name witch matches and a symbol witch containing type (but the symbol can be different).

Reply:
very, very good catch, kudos for this! 🙏